engine/modelc/src/modelimporter_gltf.cpp

```diff
@@ -1328,21 +1350,38 @@ static void LoadChannel(NodeAnimation* node_animation, cgltf_animation_channel*
     cgltf_accessor* accessor_times = channel->sampler->input;
     cgltf_accessor* accessor = channel->sampler->output;
 
+    uint32_t key_count = accessor_times->count;
+
+    // 1 for 1xf32, 4 for 4xf32
     uint32_t num_components = (uint32_t)cgltf_num_components(accessor->type);
+    // number of items per time step
+    uint32_t num_items = accessor->count / key_count;
 
     bool all_identical = true;
 
     float time_min = FLT_MAX;
     float time_max = -FLT_MAX;
 
-    KeyFrame* key_frames = new KeyFrame[accessor_times->count];
-    memset(key_frames, 0, sizeof(KeyFrame)*accessor_times->count);
-    for (uint32_t i = 0; i < accessor->count; ++i)
+    const uint32_t max_num_values = sizeof(KeyFrame::m_Value)/sizeof(float);
+    uint32_t num_values = num_items * num_components;
+    if (num_values > max_num_values)
+    {
+        dmLogWarning("Channel has input count %u and output count %u."
+                     "This yields %u items x %u components = %u"
+                     "which is larger than max count %u", (uint32_t)accessor_times->count, (uint32_t)accessor->count,
+                                                         num_items, num_components, num_values,
+                                                         max_num_values);
```

Comment:
for a grace period until we're certain about how the code should work.
Currently we can only store max 4 floats. WIll we have to increase that and/or make it dynamic number?